internal/services/synapse/synapse_pipeline_resource.go

```diff
@@ -0,0 +1,361 @@
+
+	id, err := parse.PipelineID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	client, err := synapseClient.PipelinesClient(id.WorkspaceName, environment.SynapseEndpointSuffix)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.GetPipeline(ctx, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			log.Printf("[DEBUG] Synapse Pipeline %q was not found - removing from state!", d.Id())
+			return nil
+		}
+		return fmt.Errorf("reading %s: %+v", id, err)
+	}
+
+	workspaceId := parse.NewWorkspaceID(id.SubscriptionId, id.ResourceGroup, id.WorkspaceName).ID()
+	d.Set("synapse_workspace_id", workspaceId)
+	d.Set("name", id.Name)
+
+	if props := resp.Pipeline; props != nil {
+		d.Set("description", props.Description)
+
+		parameters := flattenSynapseParameters(props.Parameters)
+		if err := d.Set("parameters", parameters); err != nil {
+			return fmt.Errorf("setting `parameters`: %+v", err)
+		}
+
+		annotations := flattenSynapseAnnotations(props.Annotations)
+		if err := d.Set("annotations", annotations); err != nil {
+			return fmt.Errorf("setting `annotations`: %+v", err)
+		}
+
+		variables := flattenSynapseVariables(props.Variables)
+		if err := d.Set("variables", variables); err != nil {
+			return fmt.Errorf("setting `variables`: %+v", err)
+		}
+
+		if activities := props.Activities; activities != nil {
+			activitiesJson, err := serializeSynapsePipelineActivities(activities)
+			if err != nil {
+				return fmt.Errorf("serializing `activities_json`: %+v", err)
+			}
+			if err := d.Set("activities_json", activitiesJson); err != nil {
+				return fmt.Errorf("setting `activities_json`: %+v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceSynapsePipelineDelete(d *schema.ResourceData, meta interface{}) error {
+	synapseClient := meta.(*clients.Client).Synapse
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	environment := meta.(*clients.Client).Account.Environment
+
+	id, err := parse.PipelineID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	client, err := synapseClient.PipelinesClient(id.WorkspaceName, environment.SynapseEndpointSuffix)
+	if err != nil {
+		return err
+	}
+
+	if _, err = client.DeletePipeline(ctx, id.Name); err != nil {
+		return fmt.Errorf("deleting %s: %+v", id, err)
+	}
+
+	return nil
+}
+
+func expandSynapseParameters(input map[string]interface{}) map[string]*artifacts.ParameterSpecification {
+	output := make(map[string]*artifacts.ParameterSpecification)
+
+	for k, v := range input {
+		output[k] = &artifacts.ParameterSpecification{
+			Type:         artifacts.ParameterTypeString,
+			DefaultValue: v.(string),
+		}
+	}
+
+	return output
+}
+
+func flattenSynapseParameters(input map[string]*artifacts.ParameterSpecification) map[string]interface{} {
+	output := make(map[string]interface{})
+
+	for k, v := range input {
+		if v != nil {
+			// we only support string parameters at this time
```

Comment:
The most minor change:

```suggestion
			// we only support string parameters at this moment
```

Reply:
fixed